include/oc_cloud.h

```diff
@@ -115,6 +133,15 @@ typedef struct oc_cloud_context_t
   int selected_identity_cred_id; /**< Selected identity cert chain. -1(default)
                                     means any*/
   bool cloud_manager;
+
+  oc_cloud_on_keepalive_response_cb_t on_keepalive_response; /**< Callback
+                                                               invoked on
+                                                               keepalive
+                                                               response */
+  uint16_t keepalive_ping_timeout; /**< Timeout for keepalive ping in seconds */
+  void *keepalive_user_data; /**< User data provided to the keepalive response
```

Comment:
Sonar will start complaining if a struct has more than 20 members (this now has 20, so it's now as big as it can be).

I would put these three in a substruct to associate them together and keep the number below 20, something like:

```C
typedef oc_cloud_keepalive_t {
  oc_cloud_on_keepalive_response_cb_t on_response; /**< Callback invoked on keepalive response */
  void* user_data; // /**< User data provided to the keepalive response callback */
  uint16_t ping_timeout; /**< Timeout for keepalive ping in seconds */
} oc_cloud_keepalive_t;
```

Also maybe we should keep the `cloud_manager` last, `sizeof(bool)==1` and one byte members tend to mess up padding if they are in the middle of the struct. (There are `uint8_t` members which have the same issue, but since this is a public struct, I guess we shouldn't mess with it too much).

Reply:
Fixed

Reply:
Fixed